python/TestHarness/schedulers/Job.py

```diff
@@ -171,6 +220,11 @@ def run(self):
         A blocking method to handle the exit status of the process object while keeping track of the
         time the process was active. When the process exits, read the output and close the file.
         """
+        if self.options.testharness_diagnostics:
+            # Give some time so that file modified time has a distinct new value.
+            time.sleep(1)
```

Comment:
Ok, I'm not sure exactly how to get this done otherwise.  The modified time goes in increments of 1 second.  If one job completes in under a second, then when it updates its modified time, it writes the same time that it picked it up at, so it doesn't record it as a file that has been written or edited.  I found code that was supposed to give me more precision on the modified time, but it didn't change anything.  Seems like the system only records up to the second for this Mac.  Suggestions?

Reply:
Do we know that these files _are_ created before we actually run (seriously, I am asking, because I don't know)? If we do, we might instead record the clock at which time the tester launched, using the-already-included clock() object (self.__start_time), and do a delta comparison with self.__end_time maybe?

Also, if we use those clock objects, you will save yourself a false-positive due to the possibility we exit this run method (due to self.__tester.shouldExecute() being False).

I am just brain-storming here...

Reply:
As far as I understand, some may be written before a test is run, while others won't exist until after the test runs.  How precise is the clock()?  Is it based on system time?  I guess I could look that up.

Reply:
For the files not created, and that we don't know about... yeah thats a toughy. But if we wanted to _assume_ anything created is going to be at clock start time... that might work. The internal clock is going to be precise enough. Try it out:

```python
from timeit import default_timer as clock
start_time = clock()
stop_time = clock()
start_time
stop_time
```
start_time and stop_time will be slightly different.

Reply:
So I'm trying to do this, but I can't quite get my head around logic that'll make it function.  Right now I'm using:
```suggestion
os.path.getmtime(fullyQualifiedFile)
```
to get the time it was modified.  I don't want to record a change in time if it wasn't modified, because the only way that I can tell if a file was changed is to compare the last-modified times.  Is there another way to flag if something was modified? The only way I had found was to to compare modified times, but if there is another way, I probably wouldn't even need to record times.

Reply:
Didn't mean to make that into a suggestion...  just meant to insert the code:
`os.path.getmtime(fullyQualifiedFile)`

Reply:
So while trying to fix this, I discovered a major bug.  Right now, if run on multiple cores, it is possible for for false positives to appear.  Looks like another job could potentially start it checking times before the previous job has finished changing the times.  So I was wondering is I should add a lock somewhere like in the RunParallel.py file:
`        if self.options.testharness_diagnostics:
            with self.locker:
                job.run()

        else:
            job.run()
`

Also, in order to avoid using sleep, I currently have:
`        if self.options.testharness_diagnostics:
            # Give some time so that file modified time has a distinct new value.
            before_check_time = clock()
            # Before the job does anything, get the times files below it were last modified
            ...
            after_check_time = clock()
            while(after_check_time - before_check_time < 1):
                after_check_time = clock()`